### PR TITLE
S3:EOS-16564: Change STs to accomodate cipher based access-secret key

### PR DESCRIPTION
**Pull request by SeagateChaDeepak**
1. Added new file(script) to create bgdelete account using cipher

**Output :**
bash-4.2$ python36 create_s3background_account_cipher.py
Account Created Successfully..
Access Key : NF9ZWTk3Nk5RN3ZjaDBaZW, Secret Key : kNTty2uFSdKz7sjsD6iNnV14l8FOwSad5XaW4fTY

http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/438/

Signed-off-by: Deepak Mahale <deepak.mahale@seagate.com>

### CODE REVIEW

File: create_s3background_account_cipher.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+import ldap.modlist as modlist
+import traceback
+from s3backgrounddelete.cortx_cluster_config import CipherInvalidToken
+from s3backgrounddelete.cortx_s3_cipher import CortxS3Cipher
+
+LDAP_USER = "cn=sgiamadmin,dc=seagate,dc=com"
+LDAP_PASSWORD = "ldapadmin"
+LDAP_URL = "ldapi:///"
+
+g_default_input_params = {
```
Comment (nileshgovande) [inline create_s3background_account_cipher.py:35]: Write one-liner comment on data and functions

Comment (nileshgovande) [inline create_s3background_account_cipher.py:35]: g_bgdel_input and g_recovery_input

File: create_s3background_account_cipher.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    else:
+        return True
+
+def disconnect_from_ldap(ldap_connection):
+    ldap_connection.unbind_s()
+
+def generate_input_params():
+    input_params = g_default_input_params
+
+    bg_access_key, bg_secret_key = generate_access_secret_keys(
```
Comment (nileshgovande) [inline create_s3background_account_cipher.py:163]: remove bg_

File: create_s3background_account_cipher.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        print("Exception : {}".format(e))
+        print("Traceback : {}".format(traceback.format_exc()))
+    finally:
+        if not ldap_connection:
+            ldap_connection.unbind()
+
+
+if __name__ == "__main__":
+    input_params = generate_input_params()
+    create_account(input_params)
```
Comment (nileshgovande) [inline create_s3background_account_cipher.py:201]: have a print saying 'adding ..... accounts'

File: create_s3background_account_cipher.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+LDAP_PASSWORD = "ldapadmin"
+LDAP_URL = "ldapi:///"
+
+g_default_input_params = {
+    '--account_name' : "s3-background-delete-svc",
+    '--account_id' : "67891",
+    '--mail' : "s3-background-delete-svc@seagate.com",
+    '--s3_user_id' : "450",
+    '--cstr_secret' : "s3backgroundsecretkey",
+    '--cstr_access' : "s3backgroundaccesskey"
```
Comment (nileshgovande) [inline create_s3background_account_cipher.py:41]: say cipher_access_key

File: create_s3background_account_cipher.py [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+                input_params['--secret_key']))
+
+        else:
+            print("Account already present.")
+
+    except Exception as e:
+        print("Exception : {}".format(e))
+        print("Traceback : {}".format(traceback.format_exc()))
+    finally:
+        if not ldap_connection:
```
Comment (nileshgovande) [inline create_s3background_account_cipher.py:195]: remove not

File: create_s3background_account_cipher.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+            ldap_connection.unbind()
+
+
+if __name__ == "__main__":
+    input_params = generate_input_params()
```
Comment (nileshgovande) [inline create_s3background_account_cipher.py:200]: take bagel or recovery as a parameter

**Review by nileshgovande — APPROVED** [review]

**Comment by nileshgovande** [comment]
1)Change file location
2)add pip3 install cryptography in init.sh after confirming with @amitwac2608

File: create_s3background_account_cipher.py [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+#!/usr/bin/env python3
```
Comment (nileshgovande) [inline create_s3background_account_cipher.py:20]: Move this up

**Review by nileshgovande — APPROVED** [review]

File: create_s3background_account_cipher.py (hunk 1 above)
Comment (SeagateChaDeepak) [reply create_s3background_account_cipher.py:35]: Done.

File: create_s3background_account_cipher.py (hunk 2 above)
Comment (SeagateChaDeepak) [reply create_s3background_account_cipher.py:163]: Done.

File: create_s3background_account_cipher.py (hunk 3 above)
Comment (SeagateChaDeepak) [reply create_s3background_account_cipher.py:201]: Have added a print statement which lists the details of created account. Kept it in alignment with the output expected by bgdelete spec tests

File: create_s3background_account_cipher.py (hunk 4 above)
Comment (SeagateChaDeepak) [reply create_s3background_account_cipher.py:41]: Done.

File: create_s3background_account_cipher.py (hunk 5 above)
Comment (SeagateChaDeepak) [reply create_s3background_account_cipher.py:195]: Done.

File: create_s3background_account_cipher.py (hunk 6 above)
Comment (SeagateChaDeepak) [reply create_s3background_account_cipher.py:200]: Done

File: create_s3background_account_cipher.py (hunk 7 above)
Comment (SeagateChaDeepak) [reply create_s3background_account_cipher.py:20]: Done

File: create_s3background_account_cipher.py (hunk 6 above)
Comment (SeagateChaDeepak) [reply create_s3background_account_cipher.py]: Done.

File: create_s3background_account_cipher.py (hunk 7 above)
Comment (SeagateChaDeepak) [reply create_s3background_account_cipher.py]: Done

File: create_s3background_account_cipher.py (hunk 1 above)
Comment (SeagateChaDeepak) [reply create_s3background_account_cipher.py]: Done.

File: s3backgrounddelete/ut/cortx_s3_config_tests.py [hunk 8]
```diff
@@ -206,49 +206,54 @@ def test_get_cortx_s3_region_failure():
     then it should throw KeyError.
     """
     with pytest.raises(KeyError):
-        config = CORTXS3Config(use_cipher = False)
+        config = CORTXS3Config()
         del config._config['cortx_s3']['default_region']
         assert config.get_cortx_s3_region()
 
 def test_get_cortx_s3_access_key_success():
```
Comment (nileshgovande) [inline s3backgrounddelete/ut/cortx_s3_config_tests.py:213]: Remove the test itself

Comment (nileshgovande) [reply s3backgrounddelete/ut/cortx_s3_config_tests.py:213]: For all

File: scripts/ldap/create_account_using_cipher.py [hunk 9]
```diff
@@ -0,0 +1,306 @@
...
+    print("[Usage:]\ncreate_s3background_account_cipher.py CreateBGDeleteAccount/CreateRecoveryAccount --ldapuser {username} --ldappasswd {passwd}")
+
+def print_create_account_results(result:dict):
+    """
+    Prints results of create account action.
+    """
+    print("AccountId = {}, CanonicalId = {}, RootUserName = root, AccessKeyId = {}, SecretKey = {}".
+        format(result['--account_id'], result['--canonical_id'], result['--access_key'], result['--secret_key']))
+
+def print_placeholder(result:dict):
```
Comment (nileshgovande) [inline scripts/ldap/create_account_using_cipher.py:262]: Remove

**Review by nileshgovande — APPROVED** [review]

**Review by knrajnambiar76 — APPROVED** [review]

File: scripts/ldap/create_account_using_cipher.py (hunk 9 above)
Comment (SeagateChaDeepak) [reply scripts/ldap/create_account_using_cipher.py:262]: Done.

File: s3backgrounddelete/ut/cortx_s3_config_tests.py (hunk 8 above)
Comment (SeagateChaDeepak) [reply s3backgrounddelete/ut/cortx_s3_config_tests.py:213]: Done.

**Pull request merged**